src/test/java/io/supertokens/test/accountlinking/api/GetUserByIdTest.java

```diff
@@ -0,0 +1,435 @@
+                    WebserverAPI.getLatestCDIVersion().get(), "");
+            JsonObject user = response.get("user").getAsJsonObject();
+
+            Set<String> phoneNumbers = new HashSet<>();
+            for (JsonElement phoneNumberElem : user.get("phoneNumbers").getAsJsonArray()) {
+                phoneNumbers.add(phoneNumberElem.getAsString());
+            }
+            assertEquals(2, phoneNumbers.size());
+            assertTrue(phoneNumbers.contains("+919876543210"));
+            assertTrue(phoneNumbers.contains("+911234567890"));
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
+
+    @Test
+    public void testThatPhoneNumberIsUnionOfLinkedAccounts3() throws Exception {
+        String[] args = {"../"};
+        TestingProcessManager.TestingProcess process = TestingProcessManager.start(args, false);
+        FeatureFlagTestContent.getInstance(process.getProcess())
+                .setKeyValue(FeatureFlagTestContent.ENABLED_FEATURES, new EE_FEATURES[]{
+                        EE_FEATURES.ACCOUNT_LINKING, EE_FEATURES.MULTI_TENANCY});
+        process.startProcess();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STARTED));
+
+        AuthRecipeUserInfo user1 = createPasswordlessUserWithPhone(process.getProcess(), "+919876543210");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user2 = createPasswordlessUserWithPhone(process.getProcess(), "+911234567890");
+        Thread.sleep(50);
+        AuthRecipeUserInfo user3 = createEmailPasswordUser(process.getProcess(), "test@example.com", "password");
+
+        AuthRecipeUserInfo primaryUser = AuthRecipe.createPrimaryUser(process.getProcess(), user1.id).user;
+        AuthRecipe.linkAccounts(process.getProcess(), user2.id, primaryUser.id);
+        AuthRecipe.linkAccounts(process.getProcess(), user3.id, primaryUser.id);
+
+        for (String userId : new String[]{user1.id, user2.id}) {
+            Map<String, String> params = new HashMap<>();
+            params.put("userId", userId);
+            JsonObject response = HttpRequestForTesting.sendGETRequest(process.getProcess(), "",
+                    "http://localhost:3567/user/id", params, 1000, 1000, null,
+                    WebserverAPI.getLatestCDIVersion().get(), "");
+            JsonObject user = response.get("user").getAsJsonObject();
+
+            Set<String> phoneNumbers = new HashSet<>();
+            for (JsonElement phoneNumberElem : user.get("phoneNumbers").getAsJsonArray()) {
+                phoneNumbers.add(phoneNumberElem.getAsString());
+            }
+            assertEquals(2, phoneNumbers.size());
+            assertTrue(phoneNumbers.contains("+919876543210"));
+            assertTrue(phoneNumbers.contains("+911234567890"));
+        }
+
+        process.kill();
+        assertNotNull(process.checkOrWaitForEvent(ProcessState.PROCESS_STATE.STOPPED));
+    }
```

Comment:
test unkonwn user id
- test with user id mapping as well - make sure that the returned user object has the user id mapped correctly, and - that the input user id has to be mapped, or unmapped (both should return the right user object)

Reply:
Done